src/lib/utils/cssSafeName.js

```diff
@@ -1,25 +1,57 @@
 /**
- * Vendored from https://github.com/thawkin3/css-safe-classname
- * MIT licensed
- * https://github.com/thawkin3/css-safe-classname/blob/991003e9d2c29e1a3c6785bf9a057d78013cc2a6/LICENSE
- * @private
+ * This script was generated by ChatGPT-4 from the prompt below. Given this
+ * code is the output of ChatGPT-4 with human revision, we do not claim copyright and this script is
+ * not covered by the same license as the rest of the repository.
+ * https://chat.openai.com/share/196ba695-0ad8-4a58-a5e6-6fd1c482093a
+ *
+ * ---------------------------------------------------------------
+ * is there a way non-safe css class names can be escaped rather than stripped? It's important to be able to keep those characters around because they have meaning, e.g. JN.1.4.3 is different from JN.1.43 - so stripping is not ok.
+ * /**
+ *  * Vendored from https://github.com/thawkin3/css-safe-classname
+ *  * MIT licensed
+ *  * https://github.com/thawkin3/css-safe-classname/blob/991003e9d2c29e1a3c6785bf9a057d78013cc2a6/LICENSE
+ *  * @private
+ *  */
+/*
+ *  const nonCssSafeCharacters = /[!"#$%&'()*+,./:;<=>?@[\\\]^`{|}~\s]/
+ *
+ *  const invalidBeginningOfClassname = /^([0-9]|--|-[0-9])/
+ *
+ *  export const cssSafeName = str => {
+ *    if (typeof str !== 'string') {
+ *      return ''
+ *    }
+ *
+ *    const replacedClassName = str.replace(
+ *      new RegExp(nonCssSafeCharacters, 'g'),
+ *      '_'
+ *    )
+ *
+ *    return invalidBeginningOfClassname.test(replacedClassName)
+ *      ? `_${replacedClassName}`
+ *      : replacedClassName
+ *  }
+ *
  */
 
-const nonCssSafeCharacters = /[!"#$%&'()*+,./:;<=>?@[\\\]^`{|}~\s]/
+const nonCssSafeCharacters = /[!"#$%&'()*+,./:;<=>?@[\\\]^`{|}~\s]/g;
 
-const invalidBeginningOfClassname = /^([0-9]|--|-[0-9])/
+const invalidBeginningOfClassname = /^([0-9]|--|-[0-9])/;
+
+function encodeCharacter(c) {
+  return '_' + c.charCodeAt(0).toString(16) + '_';
+}
 
 export const cssSafeName = str => {
   if (typeof str !== 'string') {
-    return ''
+    return '';
   }
 
-  const strippedClassname = str.replace(
-    new RegExp(nonCssSafeCharacters, 'g'),
-    ''
-  )
+  let encodedClassName = str.replace(nonCssSafeCharacters, encodeCharacter);
+
+  if (invalidBeginningOfClassname.test(encodedClassName)) {
+    encodedClassName = `_${encodedClassName}`;
```

Comment:
One could return here directly

```suggestion
    return `_${encodedClassName}`;
```

Reply:
actually the original was best I think :)

Reply:
Updated in https://github.com/nextstrain/forecasts-viz/pull/21/commits/e5f3d127f83159d05b7c511578eaef9c53f006e1